php/settings/class-setting.php

```diff
@@ -456,8 +456,23 @@ protected function register_setting() {
 	public function prepare_sanitizer( $data ) {
 
 		foreach ( $data as $slug => $value ) {
-			$setting       = $this->find_setting( $slug );
-			$data[ $slug ] = $setting->get_component()->sanitize_value( $value );
+			$setting = $this->find_setting( $slug );
+
+			$current_value = $setting->get_value();
+			$new_value     = $setting->get_component()->sanitize_value( $value );
+			/**
+			 * Filter the value before saving a setting.
+			 *
+			 * @param mixed   $new_value     The new setting value.
+			 * @param mixed   $current_value The setting current value.
+			 * @param Setting $value         The setting object.
+			 */
+			$new_value = apply_filters( "settings_save_setting_{$slug}", $new_value, $current_value, $setting );
```

Comment:
@DavidCramer 

I'm wondering if we should prepend these filters with `cloudinary` as if a namespace?